olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -560,7 +560,10 @@ client::CancellationToken AuthenticationClientImpl::SignUpHereUser(
   std::string url = settings_.token_endpoint_url;
   url.append(kUserEndpoint);
   http::NetworkRequest request(url);
-  http::NetworkSettings network_settings;
+  auto network_settings =
+      http::NetworkSettings()
+          .WithTransferTimeout(settings_.retry_settings.timeout)
+          .WithConnectionTimeout(settings_.retry_settings.timeout);
   if (settings_.network_proxy_settings) {
```

Comment:
You can merge this also above and just do:
`.WithProxySettings(settings_.proxy_settings.value_or({}));`